students/brandon_nguyen/session04/mailroom2.py

```diff
@@ -0,0 +1,143 @@
+          "Aidan Nguyen" : [4300.00,200.00,238.23],
+        }
+promptText = "\n".join(("\nWelcome to the mailroom program!\n",
+          "Please choose an options:\n\n",
+          "1 - Send a Thank You to a single donor.",
+          "2 - Create a Report.",
+          "3 - Send letters to all donors.",
+          "q - Quit",
+          ">>> "))
+
+subMenuPrompt = ("Please Chose an option:\n\nlist - To display a list of current donors.\n"
+                "1    - To enter name of donor to update.\n"
+                "q    - To exit this menu.\n"
+                ">>> ")
+#building switch case menu with dictionary
+
+
+#The script should prompt the user (you) to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”.
+
+#think about a main menu function that can be called any where.
+def show_menu(promptxt, select_dict):
+    while True:
+        menu_selected = input(promptxt)
+        if select_dict[menu_selected]() == "Exit Menu":
+            break
+
+def send_ty():
+    show_menu(subMenuPrompt, sub_menu_dict)
+
+def create_rpt():
+    """
+    This function is to print the report of donor.  
+    """
+    #Currently need more work on the format as 0.0 problem not yet solved.
+    print()
+    newList = sort_sum() #creating a new list with sorted total in reverse already for easy printing.
+    header_string = "Donor Name              |  Total Given | Num Gifts | Average Gift"
+    line = '-'*len(header_string)
+    print(header_string)
+    print(line)
+    for donor, total, num, avg in newList:
+        print(" {:<18}     $  {:>12.2f} {:>6} {:>6} {:>8.2f}".format(donor,total[0],num[0],"$",avg[0]))
+    print("\n\n")
+      
+def sort_sum():
+    #creating a new list with computed value for easy printing.
+    """
+    This function returned a sorted list of by on order amount3
+    """
+    dblist = []
+    dbSum = []
+    #quick way to convert back to list to reuse existing code.
+    for i,j in donor_db.items():
+        dblist.append((i,j))
+    for name, donation in dblist:
+        dbSum.append((name,[round(sum(donation),2)],[len(donation)],[round(sum(donation)/len(donation),2)]),)
+    return sorted(dbSum, key=lambda donor: donor[1], reverse=True)
+    
+def list_donors():
+    """
+    Basic Name of donor return in Order by First Name
+    """
+    lst = []
+    print()
+    for person in donor_db:
+        lst.append(person)
+    lst.sort()
+    for name in lst: print(name)
+    print()
+
+def update_donation():
+        input_person = input("\nPlease enter donor full names: " )
+        input_donation = float(input("Please enter donation amount for " + input_person + ":>> "))
+        #existing people add to donation
+        if input_person in donor_db:
+            new_Donation = donor_db.get(input_person)
+            new_Donation.append(input_donation)
+            donor_db[input_person] = new_Donation
+        else:
+            donor_db.update( {input_person :[input_donation]} ) 
```

Comment:
Minor: Lines 88-94 could be simplified to:
```
donor_db.set_default(input_person, []).append(input_donation)
```
Or better yet, use `defaultdict` for `donor_db` and even no need for `.set_default(...)`.